docs/state_compressor.md

```diff
@@ -0,0 +1,156 @@
+TODO: Update with final contents of README after PR #70 merged in rust-synapse-compress-state repo
```

Comment:
If we duplicate the documentation I'm rather afraid it will get out of date. Maybe we can link out or copy it as part of the build?

Reply:
Yeah I wasn't really sure what to do about the docs! I think linking them somehow would be more sensible than duplicating. I was leaving this as undone until the docs from the compressor get finalised so we know what they'll actually say!

Reply:
Makes sense! I mostly took a look cause I was curious how the rust was integrated! Figured I'd leave a couple comments while I was there!

Reply:
That PR has now merged

Reply:
I've replaced most of that with a link to the compressor documentation